feg/gateway/services/n7_n40_proxy/servicers/create_session_test.go

```diff
@@ -0,0 +1,336 @@
+	SmPolicyUrl        = "https://example.com//npcf-smpolicycontrol/v1/sm-policies/12345"
+	ActTime            = time.Unix(1634906551, 0)
+	DeactTime          = time.Unix(1634913751, 0)
+)
+
+func TestCreateSession(t *testing.T) {
+	srv, mockDb, mockN7 := newCentralSessionControllerForTest(t, false)
+
+	mockN7.On("PostSmPoliciesWithResponse", mock.Anything,
+		mock.MatchedBy(matchSmPolicyContextData(IMSI1_NOPREFIX)),
+	).Return(newSmPolicyResponse(t), nil).Once()
+
+	// no omnipresent rules
+	mockDb.On("GetOmnipresentRules").Return([]string{}, []string{}).Once()
+	// mockDb.On("GetRuleIDsForBaseNames", []string{"base_10"}).Return([]string{"base_rule_1", "base_rule_2"})
+
+	csr := newCreateSessionRequest(IMSI1)
+	response, err := srv.CreateSession(context.Background(), csr)
+	require.NoError(t, err)
+	mockN7.AssertExpectations(t)
+	assert.Equal(t, newCreateSessionResponse(IMSI1), response)
+}
+
+func TestCreateSessionTimeout(t *testing.T) {
+	srv, _, mockN7 := newCentralSessionControllerForTest(t, false)
+
+	mockN7.On("PostSmPoliciesWithResponse", mock.Anything,
+		mock.MatchedBy(matchSmPolicyContextData(IMSI1_NOPREFIX)),
+	).Return(nil, &url.Error{Err: context.DeadlineExceeded}).Once()
+
+	csr := newCreateSessionRequest(IMSI1)
+	response, err := srv.CreateSession(context.Background(), csr)
+	require.Error(t, err)
+	mockN7.AssertExpectations(t)
+	assert.Nil(t, response)
+}
+
+func TestCreateSessionErrResp(t *testing.T) {
+	srv, _, mockN7 := newCentralSessionControllerForTest(t, false)
+
+	mockN7.On("PostSmPoliciesWithResponse", mock.Anything,
+		mock.MatchedBy(matchSmPolicyContextData(IMSI1_NOPREFIX)),
+	).Return(&n7_sbi.PostSmPoliciesResponse{HTTPResponse: &http.Response{StatusCode: 400}}, nil).Once()
+
+	csr := newCreateSessionRequest(IMSI1)
+	response, err := srv.CreateSession(context.Background(), csr)
+	require.Error(t, err)
+	mockN7.AssertExpectations(t)
+	assert.Nil(t, response)
+}
+
+func TestDisableN7Response(t *testing.T) {
+	srv, mockDb, mockN7 := newCentralSessionControllerForTest(t, true)
+
+	mockDb.On("GetOmnipresentRules").Return(
+		[]string{"omni_rule_1", "omni_rule_2"}, []string{"base_10"}).Once()
+	mockDb.On("GetRuleIDsForBaseNames", []string{"base_10"}).Return([]string{"base_rule_1", "base_rule_2"})
+
+	csr := newCreateSessionRequest(IMSI1)
+	response, err := srv.CreateSession(context.Background(), csr)
+	require.NoError(t, err)
+	mockDb.AssertExpectations(t)
+	mockN7.AssertNotCalled(t, "PostSmPoliciesWithResponse")
+
+	assert.Equal(t, genSessionID(IMSI1), response.SessionId)
+	allRuleIds := []string{}
+	for _, staticRule := range response.StaticRules {
+		allRuleIds = append(allRuleIds, staticRule.RuleId)
+	}
+	assert.ElementsMatch(t, []string{"omni_rule_1", "omni_rule_2", "base_rule_1", "base_rule_2"}, allRuleIds)
+}
+
+func newCentralSessionControllerForTest(t *testing.T, disableN7 bool) (*servicers.CentralSessionController, *mockPolicyDB.PolicyDBClient, *mockN7.ClientWithResponsesInterface) {
```

Comment:
new is normally used to define something similar to a constructor.  And constructors are kind of definitions of different ways to build an object,

In this case you are just creating a default object 

```
createTestCentralSessionController
```

Reply:
change all the new ones that are not constructors of the object